Extension/src/LanguageServer/configurations.ts

```diff
@@ -752,13 +752,34 @@ export class CppProperties {
             }
 
             if (handleSquiggles) {
-                // Replace all \<escape character> with \\<character>.
+                // Replace all \<escape character> with \\<character>, except for \"
                 // Otherwise, the JSON.parse result will have the \<escape character> missing.
-                readResults = readResults.replace(/\\/g, '\\\\'); 
-                readResults = readResults.replace(/\\\\"/g, '\\"'); // Need to revert the change to \". 
-                readResults = readResults.replace(/\\\\"/g, '\\"'); // Need to do it again for \\". 
+                let newResults: string = "";
+                let lastWasBackslash: Boolean = false;
+                let lastBackslashWasEscaped: Boolean = false;
+                for (let i: number = 0; i < readResults.length; i++) {
```

Comment:
Can you factor this out into a separate function, and then add some unit tests for it?

Reply:
Can that go in a future pull request? We want to ship this fix today.